plugins/woocommerce/includes/emails/class-wc-email-customer-invoice.php

```diff
@@ -67,11 +65,7 @@ public function get_default_subject( $paid = false ) {
 		 * @return string
 		 */
 		public function get_default_heading( $paid = false ) {
-			if ( $paid ) {
-				return __( 'Invoice for order #{order_number}', 'woocommerce' );
-			} else {
-				return __( 'Your invoice for order #{order_number}', 'woocommerce' );
-			}
+			return __( 'Order #{order_number}', 'woocommerce' );
```

Comment:
```suggestion
			return __( 'Details for order #{order_number}', 'woocommerce' );
```

Via [feedback](https://github.com/woocommerce/woocommerce/pull/45395#issuecomment-1983913935) from @webdados.